src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/UseLiteralsWhereAppropriateTests.cs

```diff
@@ -169,6 +169,80 @@ End Class
             await vbTest.RunAsync();
         }
 
+        [Fact]
+        public async Task CA1802_CSharp_IntPtr_UIntPtr_NoDiagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestCode = @"
+using System;
+
+public class Class1
+{
+	internal static readonly IntPtr field1 = (nint)0;
+	internal static readonly UIntPtr field2 = (nuint)0;
+}",
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CA1802_CSharp_nint_Diagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestCode = @"
+using System;
+
+public class Class1
+{
+    internal static readonly nint field = (nint)0;
+}",
+                ExpectedDiagnostics =
+                {
+                    GetCSharpDefaultResultAt(6, 35, "field"),
+                },
+                FixedCode = @"
+using System;
+
+public class Class1
+{
+    internal const nint field = (nint)0;
+}",
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task CA1802_CSharp_nuint_Diagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                TestCode = @"
+using System;
+
+public class Class1
+{
+	internal static readonly nuint field = (nuint)0;
```

Comment:
There is a problem with the location, roslyn detects column 33 while it should be 36.

Reply:
@Evangelink there is no problem with the location. You have tab instead of four spaces, that's why.

```suggestion
    internal static readonly nuint field = (nuint)0;
```

Reply:
It's decided, I will stop my career xD

Thank you @Youssef1313 I spent 30min banging my head and decided to leave the problem for others but there is indeed no issue.

Reply:
That happens a lot to me :smile:

Reply:
You can use the `{|#0:...|}` syntax instead and never have to worry about it again.